cmd/frontend/graphqlbackend/search_results_test.go

```diff
@@ -1290,82 +1290,90 @@ func Test_ZoektSingleIndexedRepo(t *testing.T) {
 		},
 		DisableCache: true,
 	}
 	cases := []struct {
-		rev           *search.RepositoryRevisions
+		rev           string
 		wantIndexed   []*search.RepositoryRevisions
 		wantUnindexed []*search.RepositoryRevisions
 	}{
 		{
-			rev:           repoRev(""),
+			rev:           "",
 			wantIndexed:   []*search.RepositoryRevisions{repoRev("")},
 			wantUnindexed: []*search.RepositoryRevisions{},
 		},
 		{
-			rev:           repoRev("HEAD"),
+			rev:           "HEAD",
 			wantIndexed:   []*search.RepositoryRevisions{repoRev("HEAD")},
 			wantUnindexed: []*search.RepositoryRevisions{},
 		},
 		{
-			rev:           repoRev("df3f4e499698e48152b39cd655d8901eaf583fa5"),
+			rev:           "df3f4e499698e48152b39cd655d8901eaf583fa5",
 			wantIndexed:   []*search.RepositoryRevisions{repoRev("df3f4e499698e48152b39cd655d8901eaf583fa5")},
 			wantUnindexed: []*search.RepositoryRevisions{},
 		},
 		{
-			rev:           repoRev("df3f4e"),
+			rev:           "df3f4e",
 			wantIndexed:   []*search.RepositoryRevisions{repoRev("df3f4e")},
 			wantUnindexed: []*search.RepositoryRevisions{},
 		},
 		{
-			rev:           repoRev("d"),
+			rev:           "d",
 			wantIndexed:   []*search.RepositoryRevisions{},
 			wantUnindexed: []*search.RepositoryRevisions{repoRev("d")},
 		},
 		{
-			rev:           repoRev("HEAD^1"),
+			rev:           "HEAD^1",
 			wantIndexed:   []*search.RepositoryRevisions{},
 			wantUnindexed: []*search.RepositoryRevisions{repoRev("HEAD^1")},
 		},
 		{
-			rev:           repoRev("8ec975423738fe7851676083ebf660a062ed1578"),
-			wantIndexed:   []*search.RepositoryRevisions{},
-			wantUnindexed: []*search.RepositoryRevisions{repoRev("8ec975423738fe7851676083ebf660a062ed1578")},
+			rev:           "8ec975423738fe7851676083ebf660a062ed1578",
+			wantUnindexed: []*search.RepositoryRevisions{},
+			wantIndexed:   []*search.RepositoryRevisions{repoRev("8ec975423738fe7851676083ebf660a062ed1578")},
 		},
 	}
 
+	type ret struct {
+		Indexed, Unindexed []*search.RepositoryRevisions
+	}
+
 	for _, tt := range cases {
-		t.Run("classify indexed repo by commit", func(t *testing.T) {
-			filter := func(*zoekt.Repository) bool { return true }
-			indexed, unindexed, _ := zoektSingleIndexedRepo(context.Background(), z, tt.rev, filter)
-			if cmp.Diff(indexed, tt.wantIndexed) != "" {
-				t.Errorf("Got indexed repo %v, want %v", indexed, tt.wantIndexed)
-			}
-			if cmp.Diff(unindexed, tt.wantUnindexed) != "" {
-				t.Errorf("Got unindexed repo %v, want %v", unindexed, tt.wantUnindexed)
-			}
-		})
+		filter := func(*zoekt.Repository) bool { return true }
+		indexed, unindexed, err := zoektSingleIndexedRepo(context.Background(), z, repoRev(tt.rev), filter)
+		if err != nil {
+			t.Fatal(err)
+		}
+		got := ret{
+			Indexed:   indexed,
+			Unindexed: unindexed,
+		}
+		want := ret{
+			Indexed:   tt.wantIndexed,
+			Unindexed: tt.wantUnindexed,
+		}
+		if !cmp.Equal(want, got) {
+			t.Errorf("%s mismatch (-want +got):\n%s", tt.rev, cmp.Diff(want, got))
+		}
```

Comment:
nit: in your comment, is that the `cmp.Diff` not working?
```suggestion
		if diff := cmp.Diff(want, got); diff != "" {
			t.Errorf("%s mismatch (-want +got):\n%s", tt.rev, diff)
		}
```

Reply:
The previous version wasn't using the diff output. Your suggestion works, but I often use cmp.Equal. However, they are essentially the same and I see these days cmp godoc's example directly uses Diff as you do.